pkg/lightning/mydump/csv_parser.go

```diff
@@ -215,6 +215,12 @@ func (parser *CSVParser) unescapeString(input field) (unescaped string, isNull b
 			return
 		}
 		unescaped = string(decoded)
+		// we set escapedBy to empty when use base64 encoding
+		// also parser.escFlavor is not set when escapedBy is empty
+		// so directly check the unescaped is \N or not
+		if unescaped == `\N` {
```

Comment:
it's better to check `isNull` as the normal code path

```
isNull = !parser.cfg.NotNull &&
			slices.Contains(parser.cfg.Null, unescaped)
```

Reply:
since fields are already decoded and generally not quoted, I don't think similar quoting‐related logic should be apply.

Reply:
it's not about quoted. User may want to have a `\N` liternal value to be inserted into table (so need to check `cfg.NotNull`), and may choose another value to represent NULL (so need to check cfg.Null)